base/client.jl

```diff
@@ -71,6 +72,19 @@ function repl_cmd(cmd, out)
         ENV["OLDPWD"] = new_oldpwd
         println(out, pwd())
     else
+        @windows_only begin
+            if lowercase(shell[1]) == "cmd"
+                cmdbuf = IOBuffer()
+                print_cmdshell_escaped(cmdbuf, cmd.exec...)
+                cmd.exec = ["cmd"; "/s /c \"$(bytestring(cmdbuf.data))\""]
+                cmd.verbatimargument = true
+            elseif lowercase(shell[1]) == "powershell"
+                if length(cmd.exec) >= 2 && cmd.exec[2] == "--%"
+                  cmd.exec[2] = "\"--%\""
```

Comment:
this means something different: it should turn into `'"--%"'` when passed to powershell, whereas the intention was to pass `'--%'` exactly

in looking into this further however, I discovered that powershell also has a number of additional special cases
for general syntax features: http://ss64.com/ps/syntax-esc.html
and for `-command` execution: https://technet.microsoft.com/en-us/library/hh847736.aspx?f=255&MSPPError=-2147217396
these may need additional handling so that they can't be used as unintentional attack vectors.

e.g. say someone wants to write a program that does `powershell.exe -command ./julia myprog.jl -home $HOME -title $(shell_escape(user_string))`, it would be bad if the `$user_string` input could accept a string crafted to allow arbitrary (aka any) code execution or variable expansion.

note that the shell_escape character here would be "`" (powershell) not "\" (unix) or "^" (cmd).